prov/tcp/src/tcpx_cq.c

```diff
@@ -80,6 +113,8 @@ struct tcpx_xfer_entry *tcpx_xfer_entry_alloc(struct tcpx_cq *tcpx_cq,
 		return NULL;
 	}
 	tcpx_cq->util_cq.cq_fastlock_release(&tcpx_cq->util_cq.cq_lock);
+
+	tcpx_xfer_entry_reset(&tcpx_cq->buf_pools[type], xfer_entry);
```

Comment:
tcpx_xfer_entry_alloc gets called in fast path and thats why we are not doing re-init of unused fields, to improve performance. 

Do you see any issues due to this with actual tests?

Reply:
I understand your concern regarding performance, but the code currently does invalid read of uninitialized memory. 

I have a use case here where libfabric compiled with address sanitizer reports such invalid read because 
`send_entry->op_context` and `send_entry->msg_hdr.hdr.flags` are not properly and respectively initialized to NULL and 0 in `tcpx_rma_read_send_entry_fill()`. I cannot however share the source code of this test, but I guess you can reproduce similar issues with libfabric + fabtest compiled with `-fsanitize=address` (I'll try to reproduce).

I'm fine if you don't want to memset the whole structure, but `xfer_entry` is partially initialized in some locations and I bet we will face the same issue as soon as someone will add a new field into `xfer_entry` and will probably forget to initialize this new field in some places.

Reply:
We should be fine without a reproducer.  Just knowing that you hit an actual bug is sufficient.  I'll work on a fix to have the buffer pool provide zeroed memory.  Hopefully that will fix the issues that you're seeing.  Sub-structures like the msg_hdr in particular are wanting to be pre-filled once and not on each transfer.